### ISSUE TITLE
Support for dynamic child pipeline

### ISSUE THREAD

**Issue by sluongng** [mvisonneau/gitlab-ci-pipelines-exporter#196]
We have projects which have parent/child pipeline setup, where parent would generate the child's yaml dynamically and trigger child pipeline with `trigger`.

When that happen, jobs in child pipelines are available not directly under parent pipeline job lists but rather via `bridge` child pipelines. These bridges could be found using https://docs.gitlab.com/ee/api/jobs.html#list-pipeline-bridges

Correct me if Im wrong but I dont think the current exporter has support for these jobs in bridges?

**Comment by mvisonneau** [comment]
Indeed, if the (down|up)stream pipelines are not configured to be exported, you won't get their metrics. I have never used them so I am unsure how those pipelines show up in the API 🤔 

Regarding the pipeline status itself, I assume it should report correctly though.

**Comment by sluongng** [comment]
Parent / Child pipeline are very simple to reproduce:

```
# parent yaml
job-1:
  script:
    - echo 'dummy job 1'

run-child:
  trigger:
    include: path/to/child-pipeline.yml
```

```
# child yaml
job-2:
  script:
    - echo 'dummy job 2'
```

In this case:
- Listing pipeline on default branch of project will only show parent pipeline
- Listing jobs of parent pipeline will only show 'job-1'
- Listing bridge of parent pipeline will show 'run-child' which connect to a child pipeline
- Listing jobs of child pipeline will show 'job-2'

So I guess the expectation here is inside the current gitlab-ci-pipeline-exporter config, there should be a knob that turn on collecting-child-pipeline-jobs-recursively with a maximum depth of child traverse to be configurable (with a default value of '2' or '5').

When such knob is turned on, we discover child pipelines by fetching all the bridges of parent pipeline and apply the same job metric collection as we have on the parent pipelines to the discovered children.

**Comment by mvisonneau** [comment]
Thanks for the hint/explanation @sluongng! I made some tests using this repository https://gitlab.com/mvisonneau/child-pipelines and implemented the feature with 19df15c. It seems to be working fine 👌 , I enabled it be default with the possibility to disable it if necessarry. As for the maximum depth I made it infinitely traverseable for now. It looks like GitLab allows a maximum of 2 anyway.

```bash
~$ go run ./cmd/gitlab-ci-pipelines-exporter --log-level debug -c .gitlab-ci-pipelines-exporter.yml
INFO[2020-12-16T14:13:10+01:00] exporter configured                           gitlab-endpoint="https://gitlab.com" pull-environments-from-projects-interval=1800s pull-metrics-interval=10s pull-projects-from-wildcards-interval=1800s pull-rate-limit=5rps pull-refs-from-projects-interval=300s
DEBU[2020-12-16T14:13:10+01:00] pull refs from pipelines disabled, not scheduling  project-name=mvisonneau/child-pipelines
DEBU[2020-12-16T14:13:10+01:00] reading project                               project-name=mvisonneau/child-pipelines
INFO[2020-12-16T14:13:10+01:00] scheduling projects from wildcards pull       wildcards-count=0
INFO[2020-12-16T14:13:10+01:00] scheduling environments from projects pull    projects-count=1
INFO[2020-12-16T14:13:10+01:00] scheduling refs from projects pull            projects-count=1
INFO[2020-12-16T14:13:10+01:00] scheduling metrics pull                       environments-count=0 refs-count=0
INFO[2020-12-16T14:13:10+01:00] http server started                           listen-address=":8080" metrics-endpoint-enabled=true pprof-endpoint-enabled=false webhook-endpoint-enabled=false
INFO[2020-12-16T14:13:12+01:00] discovered new ref                            project-name=mvisonneau/child-pipelines ref=master ref-kind=branch
DEBU[2020-12-16T14:13:13+01:00] reading project                               project-name=mvisonneau/child-pipelines
DEBU[2020-12-16T14:13:14+01:00] listing project pipelines                     project-name=mvisonneau/child-pipelines ref=master
DEBU[2020-12-16T14:13:16+01:00] found pipeline jobs                           jobs-count=1 pipeline-id=230654434 project-name=mvisonneau/child-pipelines
DEBU[2020-12-16T14:13:16+01:00] found pipeline bridges                        bridges-count=1 pipeline-id=230654434 project-name=mvisonneau/child-pipelines
DEBU[2020-12-16T14:13:17+01:00] found pipeline jobs                           jobs-count=1 pipeline-id=230654438 project-name=mvisonneau/child-pipelines
DEBU[2020-12-16T14:13:17+01:00] found pipeline bridges                        bridges-count=1 pipeline-id=230654438 project-name=mvisonneau/child-pipelines
DEBU[2020-12-16T14:13:18+01:00] found pipeline jobs                           jobs-count=1 pipeline-id=230654443 project-name=mvisonneau/child-pipelines
DEBU[2020-12-16T14:13:19+01:00] found pipeline bridges                        bridges-count=0 pipeline-id=230654443 project-name=mvisonneau/child-pipelines
DEBU[2020-12-16T14:13:19+01:00] processing job metrics                        job-id=915231321 job-name=job1 project-name=mvisonneau/child-pipelines
DEBU[2020-12-16T14:13:19+01:00] processing job metrics                        job-id=915231332 job-name=job2 project-name=mvisonneau/child-pipelines
DEBU[2020-12-16T14:13:19+01:00] processing job metrics                        job-id=915231371 job-name=job3 project-name=mvisonneau/child-pipelines
```

**Comment by mvisonneau** [comment]
released as part of `v0.4.6`!
